src/kfactory/pcells/DCs.py

```diff
@@ -0,0 +1,148 @@
+from typing import Union
+
+from .. import KCell, autocell, kdb
+from ..generic_tech import LAYER, LayerEnum
+from ..pcells.bezier import bend_s
+from ..pcells.waveguide import waveguide
+from ..pdk import _ACTIVE_PDK
+from ..utils import Enclosure
+
+
+@autocell
+def coupler(
+    gap: float = 0.2,
+    length: float = 10.0,
+    dy: float = 5.0,
+    dx: float = 5.0,
+    width: float = 0.5,
+    layer: Union[int, LayerEnum] = LAYER.WG,
+    enclosure: Enclosure = Enclosure(),
+) -> KCell:
+    r"""Symmetric coupler.
+    Args:
+        gap: between straights in um.
+        length: of coupling region in um.
+        dy: port to port vertical spacing in um.
+        dx: length of bend in x direction in um.
+        layer: layer number or name.
+        enclosure: waveguide enclosure.
+    .. code::
+               dx                                 dx
+            |------|                           |------|
+         o2 ________                           ______o3
+                    \                         /           |
+                     \        length         /            |
+                      ======================= gap         | dy
+                     /                       \            |
+            ________/                         \_______    |
+         o1                                          o4
+    """
+    c = KCell()
+    enclosure = enclosure if enclosure is not None else Enclosure()
+    sbend = c << bend_s(
+        width=width,
+        height=((dy) / 2 - gap / 2 - width / 2),
+        length=dx,
+        layer=layer,
+        enclosure=enclosure,
+    )
+    sbend_2 = c << bend_s(
+        width=width,
+        height=((dy) / 2 - gap / 2 - width / 2),
+        length=dx,
+        layer=layer,
+        enclosure=enclosure,
+    )
+
+    sbend.transform(kdb.DTrans(0, True, 0, (gap - width + dy)))
+    sbend_2.transform(kdb.DTrans(2, False, 0, -(gap + width + dy)))
```

Comment:
Transforming them to then connect them? That shouldn't be necessary

Reply:
Edited